node/src/types/block.rs

```diff
@@ -936,6 +939,18 @@ impl Display for BlockHeaderWithMetadata {
     }
 }
 
+impl Item for BlockHeaderWithMetadata {
+    type Id = u64;
+    type ValidationError = BlockHeaderWithMetadataValidationError;
+    const TAG: Tag = Tag::BlockHeaderAndFinalitySignaturesByHeight;
+    const ID_IS_COMPLETE_ITEM: bool = false;
+
+    fn id(&self) -> Result<Self::Id, Self::ValidationError> {
+        validate_block_header_and_finality_signatures(&self.block_header, &self.block_signatures)?;
```

Comment:
Isn't it a case that every node can be configured with a different FTT and still participate in the same network? If my local node is configured with stronger safety requirements then the weight of finality signatures that I expect will have to greater than sender's. In that case we might start disconnecting from peers we don't have to.

I said it in the past and will only repeat here - it should be a responsibility of the _requesting_ node to collect signatures until it satisfies his local safety requirements.

Reply:
This function doesn't check that it's _enough_ finality signatures, just that the ones received are valid and belong to this block.

Reply:
Ah sorry, my mistake. There's another one that does it then.

Reply:
Yes, but at least that's not introduced in this PR. :sweat_smile: